theano/tensor/elemwise.py

```diff
@@ -696,7 +696,7 @@ def as_scalar(t):
 
             scalar_inputs = list(map(as_scalar, inputs))
             scalar_ograds = list(map(as_scalar, ograds))
-            scalar_igrads = self.scalar_op.grad(scalar_inputs, scalar_ograds)
+            scalar_igrads = self.scalar_op.L_op(scalar_inputs, outputs, scalar_ograds)
```

Comment:
This part isn't good. I think you should do something like:

scalar_outputs = self.scalar_op.make_node(
            *[get_scalar_type(dtype=i.type.dtype).make_variable()
              for i in inputs])

and replace outputs by scalar_outputs in this line. More change bellow.